src/__tests__/Document.test.ts

```diff
@@ -856,7 +856,133 @@ describe('validate', () => {
 			});
 			const document = new DocumentSubclass(schema, { data: { prop1: 'foo' } });
 
-			const expected = new Map([['prop1', 'Test error message']]);
+			const expected = new Map([['prop1', ['Test error message']]]);
+			expect(await document.validate()).toEqual(expected);
+		});
+
+		test('return error at property & unravel nested errors if schemaType validation fails', async () => {
+			const passingEmbeddedObjectSchema: SchemaDefinition = {
+				prop1: [{ type: 'number', path: '7', required: true }],
+				prop2: { type: 'number', path: '8', required: true },
+			};
+			const failingEmbeddedObjectSchema: SchemaDefinition = {
+				prop1: [{ type: 'number', path: '9', required: true }],
+				prop2: { type: 'number', path: '10', required: true },
+			};
+
+			const passingEmbeddedObjectDocumentArraySchema: SchemaDefinition = {
+				prop1: [{ type: 'string', path: '11', required: true }],
+				prop2: { type: 'number', path: '12', dbDecimals: 2, required: true },
+			};
+			const failingEmbeddedObjectDocumentArraySchema: SchemaDefinition = {
+				prop1: [{ type: 'string', path: '13', required: true }],
+				prop2: { type: 'number', path: '14', dbDecimals: 2, required: true },
+			};
+
+			const passingNestedSchema = new Schema({
+				prop1: { type: 'string', path: '15', required: true },
+				prop2: { type: 'number', path: '16', dbDecimals: 2, required: true },
+			});
+			const failingNestedSchema = new Schema({
+				prop1: [{ type: 'string', path: '17', required: true }],
+				prop2: { type: 'number', path: '18', dbDecimals: 2, required: true },
+			});
+
+			const passingNestedSchemaDocumentArraySchema = new Schema({
+				prop1: [{ type: 'string', path: '19', required: true }],
+				prop2: [{ type: 'number', path: '20', dbDecimals: 2, required: true }],
+			});
+			const failingNestedSchemaDocumentArraySchema = new Schema({
+				prop1: [{ type: 'string', path: '21', required: true }],
+				prop2: { type: 'number', path: '22', dbDecimals: 2, required: true },
+			});
+			const definition: SchemaDefinition = {
+				passingProp: { type: 'string', path: '1', required: true },
+				failingProp: { type: 'number', path: '2', dbDecimals: 2, required: true },
+				passingArray: [{ type: 'string', path: '3', required: true }],
+				failingArray: [{ type: 'string', path: '4', required: true }],
+				passingNestedArray: [[{ type: 'string', path: '5', required: true }]],
+				failingNestedArray: [[{ type: 'string', path: '6', required: true }]],
+				passingEmbeddedObject: passingEmbeddedObjectSchema,
+				failingEmbeddedObject: failingEmbeddedObjectSchema,
+				passingEmbeddedObjectDocumentArray: [passingEmbeddedObjectDocumentArraySchema],
```

Comment:
Similarly here, I wouldn't refer to this as an "embedded object".

Reply:
Making that change. Leaving `Object` so it can differentiate from the use of `new Schema`.